src/shared/remoteSourceTrackingService.ts

```diff
@@ -500,11 +500,38 @@ export class RemoteSourceTrackingService extends ConfigFile<RemoteSourceTracking
 
   // Return a tracked element as MemberRevision data.
   private getSourceMember(key: string): Optional<MemberRevision> {
-    return this.getSourceMembers()[key];
+    const sourceMembers = this.getSourceMembers();
+    let sm = sourceMembers[key];
+    if (!sm) {
+      // Get all SourceMember keys in maxRevision, then iterate over them
+      // and compare their decoded value with the decoded key.
+      Object.keys(sourceMembers).some((memberKey) => {
```

Comment:
I didn't love the repetitiveness, so I did a refactor (which also helped me follow the logic better).  The UT were great to make that possible

https://github.com/forcedotcom/source-tracking/pull/460

If you don't like it, it's also approved as-is

Reply:
I'll merge it